### PR TITLE
Generated extension accessor can configure DeferredConfigurable extensions

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

File: src/test/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchemaAccessorsIntegrationTest.kt [hunk 1]
```diff
@@ -10,10 +10,46 @@ import org.junit.Assert.assertFalse
...
+            dependencies {
+                "api"("org.apache.commons:commons-lang3:3.5")
+            }
+
+        """.replaceIndent())
+
+        withAutomaticAccessors()
+        build("generatePom")
+
+        val pom = existing("build/publications/mavenJavaLibrary/pom-default.xml").readText(UTF_8)
```
Comment (bamboo) [inline src/test/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchemaAccessorsIntegrationTest.kt]: `UTF_8` is already the [default](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/read-text.html).

File: src/main/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchema.kt [hunk 2]
```diff
@@ -98,25 +98,43 @@ fun loadMultiProjectSchemaFrom(file: File) =
...
 }
 
+private
+fun extensionAccessorFor(name: String, type: String): String? =
+    if(isLegalExtensionName(name))
+        """
+            /**
+             * Retrieves the [$name][$type] project extension.
+             */
+            fun Project.`$name`(): $type =
```
Comment (bamboo) [inline src/main/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchema.kt]: We should take this opportunity to strongly consider #297.

File: src/test/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchemaAccessorsIntegrationTest.kt [hunk 3]
```diff
@@ -10,10 +10,46 @@ import org.junit.Assert.assertFalse
...
+                publications.create<MavenPublication>("mavenJavaLibrary") {
+                    from(components["java"])
+                }
+            }
+
+            dependencies {
+                "api"("org.apache.commons:commons-lang3:3.5")
+            }
+
+        """.replaceIndent())
```
Comment (bamboo) [inline src/test/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchemaAccessorsIntegrationTest.kt]: `replaceIndent` is unnecessary here.

File: src/test/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchemaAccessorsIntegrationTest.kt [hunk 4]
```diff
@@ -10,10 +10,46 @@ import org.junit.Assert.assertFalse
...
+
+        withAutomaticAccessors()
+        build("generatePom")
```
Comment (bamboo) [inline src/test/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchemaAccessorsIntegrationTest.kt]: I would add a blank line here, before the call to `build(...)` to visually separate the _arrange_ section from the _act_ section.

File: src/main/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchema.kt [hunk 5]
```diff
@@ -98,25 +98,43 @@ fun loadMultiProjectSchemaFrom(file: File) =
...
         if (name !in extensions) {
-            accessorFor(name, type, "convention", "convention.getPluginByName")?.let(action)
+            conventionAccessorFor(name, type)?.let(action)
         }
     }
 }
 
+private
+fun extensionAccessorFor(name: String, type: String): String? =
+    if(isLegalExtensionName(name))
```
Comment (bamboo) [inline src/main/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchema.kt]: There's a space missing between the `if` and the opening paren.

File: src/test/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchemaAccessorsIntegrationTest.kt [hunk 6]
```diff
@@ -10,10 +10,46 @@ import org.junit.Assert.assertFalse
 import org.junit.Test
 
 import java.io.File
+import java.nio.charset.StandardCharsets.UTF_8
 
 class ProjectSchemaAccessorsIntegrationTest : AbstractIntegrationTest() {
 
     @Test
+    fun `can configure deferred configurable extension`() {
+        withBuildScript("""
```
Comment (bamboo) [inline src/test/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchemaAccessorsIntegrationTest.kt]: I'd add a blank line between the back-quoted function name and `withBuildScript(...)`.

File: src/main/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchema.kt (hunk 2 above)
Comment (eskatos) [inline src/main/kotlin/org/gradle/script/lang/kotlin/accessors/ProjectSchema.kt]: Right! Let's do this as a subsequent change.

**Comment by eskatos** [comment]
@bamboo I addressed your feedback, could you have another look?

**Pull request merged**